How to Deploy a Shiny App.md

rmccreath opened the pull request:
# Pull Request Details

**Issue Number**: Closes #44 

**Type**: Documentation

## Description of the Change

Adding guidance for Shiny deployment to sit alongside other 'information sharing' content, covering key considerations for sharing and the deployment process for Shiny apps onto shinyapps.io.

## Release Notes

* Added Shiny app deployment guidance

Moohan commented on Shiny/Publish a Shiny App.md:
```diff
@@ -0,0 +1,42 @@
+
+If something doesn’t work when your app is already in the server (but works locally) you can check the app logs using the command:  
+
+`rsconnect::showLogs(appName=”... use app name here ...”)`
```
```suggestion
`rsconnect::showLogs(appName = "<app name>")`
```

Moohan commented on Shiny/Publish a Shiny App.md:
```diff
@@ -0,0 +1,42 @@
+
+1. Fill in this form for each app instance. As this is a shared service, we need to know which apps belong to PHS, so we can contact app authors if there are any changes on the shiny.io server, and to allow for organisational oversight of our published content.
+    - As the shiny.io server is used by different public sector organisations we follow a naming convention to make things clearer, so your app will need to be named following this format: "phs-descriptive-name".
+2. Run the “deploying_app” script you can find in the files section of this channel.
```
```suggestion
2. Run the "deploying_app" script you can find in the files section of this channel.
```

Moohan commented on Shiny/Publish a Shiny App.md:
```diff
@@ -0,0 +1,42 @@
+It is possible to password-protect Shiny apps. To align with governance and data protection, the current agreement is that any protection should only be done only for a limited amount of time before publication for _testing_, _quality assurance_ and _pre-release access_ purposes. Password protection is set up by the app author using the package `{shinymanager}`. The steps are: 
+
+1. Add the package `{shinymanager}` to your global script.  
+2. Add a function called “secure_app” at the beginning of the ui script. 
```
```suggestion
2. Add a function called "secure_app" at the beginning of the ui script. 
```

Moohan commented on Shiny/Publish a Shiny App.md:
```diff
@@ -0,0 +1,42 @@
+
+> [!NOTE]
+> Before deploying an app, check that it works locally and that everything the app needs is in the app folder as through the upload process you basically upload your app folder to the server.
+> All file paths need to be relative rather than absolute as well, as the app in the server won’t have access to PHS’ file structure.  
```
```suggestion
> All file paths need to be relative rather than absolute as well, as the app in the server won't have access to PHS' file structure.  
```

Moohan replied:
```suggestion
> All file paths need to be relative rather than absolute as well, as the app in the server won’t have access to PHS' file structure.  
```

rmccreath commented:
Feedback from @juliaromero-phs 

> **Deploying a Shiny app**
> 
> 1. Deploying a Shiny app - Mention that the code mentioned should sit in a separate script and just needs to be ran like a "normal" script to deploy.
> 2. Troubleshooting - If dashboard doesn't deploy, Profile code using the [profvis package](https://cran.r-project.org/web/packages/profvis/index.html.)
> 3. Troubleshooting - If a dashboard is large, and code has been optimised with profiling, the dashboard might need more memory on the Shiny server.

rmccreath commented:
@juliaromero-phs 
1. Updated the text to describe purpose and not to include within the app codebase.
2. Great suggestion, added in for troubleshooting.
3. I think keeping the final statement as getting in touch with other users/admins will prevent users from defaulting to more memory as the solution. It's a limited resource available and gives admins space to ensure performance is covered.

Thanks again for your feedback. Could you review again/approve if you think it's ready for sharing?

JFix89 commented on Information Sharing/Deploying a Shiny App.md:
```diff
@@ -0,0 +1,102 @@
+
+`rsconnect::showLogs(appName = "<app name>")`
+
+If performance is a concern, this should be notable during development but can become more of a problem after deployment with limited resources and sessions covering multiple users. The [`{profvis}`](https://profvis.r-lib.org/index.html) is an interactive tool for visualising code profiling data and can be useful for identifying where bottlenecks or performance issues are within an application.
```
```suggestion
If performance is a concern, this should be notable during development but can become more of a problem after deployment with limited resources and sessions covering multiple users. The [`{profvis}`](https://profvis.r-lib.org/index.html) package is an interactive tool for visualising code profiling data and can be useful for identifying where bottlenecks or performance issues are within an application.
```

JFix89 reviewed (changes requested):
Also looks good to me overall, thanks again!

Moohan commented on Information Sharing/Deploying a Shiny App.md:
```diff
@@ -0,0 +1,128 @@
+
+There are some key points to consider before deploying an app:
+
+- **Data**: Ensure that the data used in the app is not confidential and can be shared publicly. If you are unsure, contact the [PHS Stats Gov team](mailto:phs.statsgov@phs.scot).
```
I think it needs to be made extra clear here that data can't be confi whether it is available for users to view - or even stored in the background at any point.

rmccreath replied:
Agreed, updated with something more specific.

Moohan reviewed (approved):
I sent a message to Russell about naming conventions and how that might link up in a Posit Connect future but it's more something to think about, not a change to be made here.

Happy with this as far as I can see - Note I have never used the PHS Shiny app deployment process, it would be good to have at least one person who has to review this.
